dgv/metrics/task_functions.py

```diff
@@ -126,91 +271,63 @@ def get_dict(df, obj_property):
     return arr
 
 
-def get_date(a_date):
-    return datetime.strptime(a_date, "[%d/%b/%Y:%H:%M:%S.%f]").strftime("%Y-%m-%d")
-
-
-def search_pattern(patterns, value, type_object):
-    for pattern in patterns:
-        if pattern in value:
-            slug = value.replace(pattern, "").split("/")[0].replace(";", "")
-            return slug, True, type_object
-    return None, False, None
-
-
-def search_pattern_resource_static(pattern, value, type_object):
-    if pattern in value:
-        slug = f"https://static.data.gouv.fr{value}".replace(";", "")
-        return slug, True, type_object
-    return None, False, None
-
-
-def get_info(parsed_line):
-    languages = ["fr", "en", "es"]
-    patterns_datasets = [f"/{lang}/datasets/" for lang in languages]
-    patterns_reuses = [f"/{lang}/reuses/" for lang in languages]
-    patterns_organizations = [f"/{lang}/organizations/" for lang in languages]
-    patterns_resources_id = [f"/{lang}/datasets/r/" for lang in languages]
-    pattern_resources_static = "/resources/"
-    slug_line = None
-    found = False
-
+def get_info(parsed_line: str) -> Tuple[Optional[str], Optional[str], Optional[str]]:
+    """
+    Retrieve information related to the datasets, organisation or resources
+    from the anonymised HAProxy logs.
+
+    Args:
+        parsed_line (str): HAProxy line to parse
+    Exemple:
+        parsed_line = '2024-11-13T00:00:23.927326+01:00 slb-04 haproxy[260742]: 127.0.0.1:37959 '
+                      '[13/Nov/2024:00:00:23.908] DATAGOUVFR_RGS~ DATAGOUVFR_NEWINFRA/dataweb-06 '
+                      '0/0/2/16/+18 302 +684 - - --NN 222/189/4/1/0 0/0 '
+                      '"GET /fr/datasets/r/ee16d126-af0f-4b3b-84d3-080ef8bc0abd HTTP/1.1"'
+        Output:
+            slug_line = "ee16d126-af0f-4b3b-84d3-080ef8bc0abd"
+            type = "resources"
+            segment = "fr"
+    """
+    static_slug_line, static_obj_type, static_segment = (None, None, None)
     if (
+        # DATAGOUVFR_RGS: service behind the RGS certificate so www.data.gouv.fr
         "DATAGOUVFR_RGS~" in parsed_line
         and '"GET' in parsed_line
-        and ('302' in parsed_line or '200' in parsed_line)
+        and ("302" in parsed_line or "200" in parsed_line)
```

Comment:
Good question.
Here is an example of log:
```
2024-12-17T00:00:19.439613+01:00 slb-04 haproxy[234222]: X.X.X.X[17/Dec/2024:00:00:19.431] DATAGOUVFR_RGS~ DATAGOUVFR/XX 0/0/1/3/+4 302 +415 - - --NN 123/4/5/0/0 0/0 "GET /fr/datasets/r/1f930c9a-5c15-4aea-a961-baa98f176dcb HTTP/1.1"
```
This is the original logic, I did not change it but I think it should be safe since looking at the log structure it would be very unlikely to find another lone `302` or `200`. `parsed_line` is the split of the log line on spaces.

Reply:
It also raises a question. Should we really include the `302` code?

A 302 will redirect to another page that until it raises a 200 (or an error) right?
So won't we count twice the traffic ?
Example:
```
2024-11-13T23:09:16.940506+01:00 slb-04 haproxy[1234]: X.X.X.X [13/Nov/2024:23:09:16.929] DATAGOUVFR_RGS~ DATAGOUVFR/prod 0/0/1/2/+3 302 +432 - - --NN 313/227/5/3/0 0/0 "GET /datasets/57868ea9a3a7295d371adcfe/ HTTP/1.1"

2024-11-13T23:09:17.517371+01:00 slb-04 haproxy[1234]: X.X.X.X [13/Nov/2024:23:09:17.039] DATAGOUVFR_RGS~ DATAGOUVFR/prod 0/0/2/324/+234 200 +1234 - - --NR 315/229/5/2/0 0/0 "GET /fr/datasets/logements-sociaux-subventionnes-dans-les-communes-ayant-moins-de-25-de-logements-sociaux/ HTTP/1.1"
```
What do you think?

Reply:
Regarding the first message: I'm confused, isn't `parsed_line` the line itself as a string? In which case it's not unlikely to find such substrigs within right?
About the `302` that's a good question, maybe rather for end-users of the metrics. I summon @maudetes 🪄

Reply:
About the first message:

Amazing catch!
Indeed, I removed the `.split()` for a more robust regex to avoid applying the pattern logic to each element of the log. But doing so reduced the robustness of:
```
"DATAGOUVFR_RGS~" in parsed_line
and '"GET' in parsed_line
and ("302" in parsed_line or "200" in parsed_line)
```
I can replace it with:
```
" DATAGOUVFR_RGS~ " in parsed_line
and ' "GET ' in parsed_line
and (" 302 " in parsed_line or " 200 " in parsed_line)
```
Seems basic but this should work out. Or I can apply the `.split()` back. What would you prefer?